src/exif.c

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/internal.h"
+
+#include <stdint.h>
+#include <string.h>
+
+avifResult avifExtractExifTiffHeaderOffset(const avifRWData * exif, uint32_t * exifTiffHeaderOffset)
+{
+    *exifTiffHeaderOffset = 0;
+    if (exif->size < 4) {
+        // Can't even fit the TIFF header, something is wrong
+        return AVIF_RESULT_INVALID_EXIF_PAYLOAD;
+    }
+
+    const uint8_t tiffHeaderBE[4] = { 'M', 'M', 0, 42 };
+    const uint8_t tiffHeaderLE[4] = { 'I', 'I', 42, 0 };
+    for (; *exifTiffHeaderOffset < (exif->size - 4); ++*exifTiffHeaderOffset) {
+        if (!memcmp(&exif->data[*exifTiffHeaderOffset], tiffHeaderBE, sizeof(tiffHeaderBE))) {
+            break;
+        }
+        if (!memcmp(&exif->data[*exifTiffHeaderOffset], tiffHeaderLE, sizeof(tiffHeaderLE))) {
```

Comment:
Nit: It's safe to use 4 instead of `sizeof(tiffHeaderBE)` and `sizeof(tiffHeaderLE)` because the size of those two arrays is fixed by the standard. Also this function already uses 4 in three places.